Server/src/RedisManager.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2024 Microsoft, Inc. and others.  All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ */
+
+#include "RedisManager.h"
+
+
+/*********************************************************************//**
+ * Constructor for class
+ ***********************************************************************/
+RedisManager::RedisManager() {
+    exit_ = false;
+}
+
+/*********************************************************************//**
+ * Constructor for class
+ ***********************************************************************/
+RedisManager::~RedisManager() {
+}
+
+
+/*********************************************************************
+ * Setup for this class
+ *
+ * \param [in] logPtr     logger pointer
+ * \param [in] cfgPtr     config pointer
+ ***********************************************************************/
+void RedisManager::Setup(Logger *logPtr, Config *cfgPtr) {
+    logger = logPtr;
+    if (!cfgPtr->redis_multiAsic) {
+        if (!swss::SonicDBConfig::isInit()) {
+            swss::SonicDBConfig::initialize();
+        }
+    } else {
+        if (!swss::SonicDBConfig::isGlobalInit()) {
+            swss::SonicDBConfig::initializeGlobalConfig();
+        }
+    }
+    stateDb_ =  std::make_shared<swss::DBConnector>(BMP_DB_NAME, 0, false);
+    separator_ = swss::SonicDBConfig::getSeparator(BMP_DB_NAME);
+}
+
+
+
+/**
+ * Get Key separator for deletion
+ *
+ * \param [in] N/A
+ */
+std::string RedisManager::GetKeySeparator() {
+    return separator_;
+}
+
+
+/**
+ * WriteBMPTable
+ *
+ * \param [in] table            Reference to table name
+ * \param [in] key              Reference to various keys list
+ * \param [in] fieldValues      Reference to field-value pairs
+ */
+bool RedisManager::WriteBMPTable(const std::string& table, const std::vector<std::string>& keys, const std::vector<swss::FieldValueTuple> fieldValues) {
+
+    if (enabledTables_.find(table) == enabledTables_.end()) {
+        LOG_INFO("RedisManager %s is disabled", table.c_str());
+        return false;
+    }
+    std::unique_ptr<swss::Table> stateBMPTable = std::unique_ptr<swss::Table>(new swss::Table(stateDb_.get(), table));
+    std::string fullKey;
+    for (const auto& key : keys) {
+        fullKey += key;
+        fullKey += separator_;
+    }
+    fullKey.erase(fullKey.size() - 1);
+
+    DEBUG("RedisManager WriteBMPTable key = %s", fullKey.c_str());
+
+    stateBMPTable->set(fullKey, fieldValues);
```

Comment:
>fieldValues [](http://example.com/codeflow?start=32&length=11)

Suggest to log this values or size of it for future tracebility in complex scenarios

Reply:
added more debug log into MsgBusImpl_redis class for more constructive looking, which triggers this WriteBMPTable()